yt_dlp/extractor/joqrag.py

```diff
@@ -0,0 +1,67 @@
+import urllib.parse
+
+from .common import InfoExtractor
+from ..utils import clean_html, urljoin
+
+
+class JoqrAgIE(InfoExtractor):
+    IE_DESC = '超!A&G+ 文化放送 Nippon Cultural Broadcasting, Inc. (JOQR)'
+    _VALID_URL = [r'https?://www\.uniqueradio\.jp/agplayer5/player\.php',
+                  r'https?://www\.uniqueradio\.jp/agplayer5/inc-player-hls\.php',
+                  r'https?://(?:www\.)?joqr\.co\.jp/ag/',
+                  r'https?://(?:www\.)?joqr\.co\.jp/qr/(?:agdailyprogram|agregularprogram)/']
+    _TESTS = [{
+        'url': 'https://www.uniqueradio.jp/agplayer5/player.php',
+        'info_dict': {
+            'id': 'live',
+            'title': str,
+            'channel': '超!A&G+',
+            'description': str,
+            'live_status': 'is_live',
+        },
+        'params': {
+            'skip_download': True,
+            'ignore_no_formats_error': True,
+        },
+    }, {
+        'url': 'https://www.uniqueradio.jp/agplayer5/inc-player-hls.php',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.joqr.co.jp/ag/article/103760/',
+        'only_matching': True,
+    }, {
+        'url': 'http://www.joqr.co.jp/qr/agdailyprogram/',
+        'only_matching': True,
+    }, {
+        'url': 'http://www.joqr.co.jp/qr/agregularprogram/',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = 'live'
+
+        metadata = self._download_webpage(
+            'https://www.uniqueradio.jp/aandg', video_id,
+            note='Downloading metadata', errnote='Failed to download metadata')
+        title = clean_html(urllib.parse.unquote_plus(
+            self._search_regex(r'var\s+Program_name\s*=\s*["\']([^"\']+)["\']', metadata, 'program title')))
+        desc = clean_html(urllib.parse.unquote_plus(
+            self._search_regex(r'var\s+Program_text\s*=\s*["\']([^"\']+)["\']', metadata, 'program description')))
```

Comment:
since it's basically the same thing for both, you could maybe add:
```python3
    def _extract_metadata(self, variable, html, name):
        text = self._search_regex(fr'var\s+{variable}\s*=\s*["\']([^"\']+)["\']', html, name)
        if text:
            return clean_html(urllib.parse.unquote_plus(text))
```
then you can do:
```suggestion
        title = self._extract_metadata('Program_name', metadata, 'program title')
        desc = self._extract_metadata('Program_text', metadata, 'program description')
```
and then that's not fatal if the regex doesn't find the metadata

Reply:
need to add `fatal=False` to the `_search_regex()` if you want it to be non-fatal w/ warnings

or else this for no warnings
```py
    def _extract_metadata(self, variable, html, name):
        return clean_html(urllib.parse.unquote_plus(self._search_regex(
            rf'var\s+{variable}\s*=\s*["\']([^"\']+)["\']', html, name, default=''))) or None
```

Reply:
either way, it should definitely be non-fatal though